### PR TITLE
RAINCATCH-451: Update to use new simple store module

### PR DESCRIPTION
**Pull request by JameelB**
## What
Updated server.js to add in an id field, required by the new raincatcher-simple-store module, to the message object upon creation.

## JIRA Ticket
https://issues.jboss.org/browse/RAINCATCH-451

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-  , config = require('./config');
+var sync = require('fh-wfm-sync/lib/server'),
+  config = require('./config');
 
 
 module.exports = function(mediator, app, mbaasApi) {
   sync.init(mediator, mbaasApi, config.datasetId, config.syncOptions);
+
+  // Add an id field required by the new simple store module to the workorder object that will be created
+  mediator.subscribe('wfm:cloud:workorders:create', function(workorderToCreate, ts) {
```
Comment (nialldonnellyfh) [inline lib/server.js]: I think it is `wfm:cloud:workorder:create`.

These topics should be generated or moved to constants.

Comment (nialldonnellyfh) [inline lib/server.js]: `ts` should really be called `mediatorTopicIdentifier`.

File: lib/server.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+    workorderToCreate.id = ts;
```
Comment (nialldonnellyfh) [inline lib/server.js]: `ts` is not guaranteed to be unique so I would advise to generate an ID here.

File: lib/server.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+    mediator.publish('wfm:cloud:data:workorders:create', workorderToCreate);
```
Comment (nialldonnellyfh) [inline lib/server.js]: After this topic you will need to publish the `done` topic for `wfm:cloud:workorders:create` also.

**Comment by vchepeli** [comment]
@JameelB, @nialldonnellyfh think do we need some unit tests for this?

File: lib/server.js [hunk 4]
```diff
@@ -1,9 +1,48 @@
...
 
 
 module.exports = function(mediator, app, mbaasApi) {
   sync.init(mediator, mbaasApi, config.datasetId, config.syncOptions);
+
+  mediator.subscribe(config.topicPrefix + config.datasetId + ':create', function(workorderToCreate, mediatorTopicIdentifier) {
+    // Add an id field required by the new simple store module to the workorder object that will be created
+    workorderToCreate.id = shortid.generate();
+
+    mediator.request(config.topicPrefix + 'data:' + config.datasetId + ':create', workorderToCreate, {uid: workorderToCreate.id})
```
Comment (nialldonnellyfh) [inline lib/server.js]: It would be good to add a small utility function to generate the topics.

File: lib/config.js [hunk 5]
```diff
@@ -8,5 +8,6 @@ module.exports = {
     "sync_frequency" : 5,
     "storage_strategy": "dom",
     "do_console_log": false
-  }
+  },
+  topicPrefix: 'wfm:cloud:'
```
Comment (nialldonnellyfh) [inline lib/config.js]: This should probably be called "cloudTopicPrefix" if it is referencing wfm:cloud.

**Pull request merged**